### ISSUE TITLE
Locale not preserved on hosted login pages

### ISSUE THREAD

**Issue by mooreds** [FusionAuth/fusionauth-issues#1713]
### Description 

Locale is not preserved when using the hosted login pages and clicking a link. Also not preserved in form submits.

### Affects versions
1.36.4

### Steps to reproduce

1. Stand up a new FusionAuth install
2. Add a french theme. I used the locale file here: https://raw.githubusercontent.com/FusionAuth/fusionauth-localization/master/theme/messages_fr.properties
3. Make sure you save the theme.
4. Update the tenant to use the new theme.
5. Open an incognito window.
6. Go to the authorize page in the install
7. Append `&locale=fr` to the url, it loads up fine and shows you French.
8. Click a link or submit the login form without a password.

You are shown the English messages.

### Expected behavior

Keep the locale persistent. 

### Community guidelines
All issues filed in this repository must abide by the [FusionAuth community guidelines](https://fusionauth.io/community/forum/topic/1000/code-of-conduct).

### Additional context

Originally mentioned here by @glen-84: https://github.com/FusionAuth/fusionauth-issues/issues/1693#issuecomment-1105553050

May need to update the locale docs here if behavior has changed: https://fusionauth.io/docs/v1/tech/core-concepts/localization-and-internationalization

**Comment by robotdan** [comment]
So if you append `&locale=fr` to the URL it works. The documented way for an end user to select a locale is to use the Locale selector on the themed page, if you change that what happens?

**Comment by mooreds** [comment]
Ah, great. It works with the dropdown.

I see a cookie dropped `fusionauth.locale` and the locale changes.

So I see a few possible changes:

* we document that the only supported way to specify the locale is to use the drop down component. 
* Perhaps we make it a bit smarter so it looks at the locale url parameter when it is first loaded.
* we could document the fact that the `fusionauth.locale` cookie needs to be set for locale preferences to work, and let folks customizing their themes drop this cookie however they want.

**Comment by mooreds** [comment]
Right now the documentation is definitely incorrect as it specifies that the `locale` parameter controls. I think I looked that up, but must have gotten it wrong. FusionAuth uses the https://github.com/prime-framework/prime-mvc/blob/master/src/main/java/org/primeframework/mvc/locale/DefaultLocaleProvider.java#L53 and that definitely doesn't consult the locale parameter. I don't know where I got that info from.

**Comment by robotdan** [comment]
Ok, perhaps that used to work and it doesn't any longer. We can use this issue to investigate that. The user would also have the option to use JS to take the value from the URL use it to set the local selector.

**Comment by glen-84** [comment]
Note that:

1. The value of the locale selector is correctly set using the query string parameter.
2. The locale selector is outside of the form, so its value is not submitted. I suppose it's relying on the cookie.
3. The locale selector only sets a cookie when the selection changes. It should set the cookie regardless.

The quickest fix would probably be to set the `fusionauth.locale` cookie on page load in addition to locale selection change. In this way, it's not necessary to append the locale parameter to all links/actions on the page.

**Comment by glen-84** [comment]
@robotdan

If I set a `locale` query string parameter, I expect that to be persisted. Since a cookie is used for this purpose, I expect a cookie to be set when loading a hosted form with a `locale` parameter.

Currently, the parameter is used once to translate the page, and then forgotten. The cookie is only set when selecting a different locale. It should also be set when loading the page. Does that make sense?

**Comment by mooreds** [comment]
What Glen said :). Thanks Glen.

**Comment by robotdan** [comment]
Is this just a doc issue, or is there a code change required in FusionAuth?

**Comment by mooreds** [comment]
@robotdan I'd say Glen captures the requirement, which appears to be a code change:

> If I set a locale query string parameter, I expect that to be persisted. Since a cookie is used for this purpose, I expect a cookie to be set when loading a hosted form with a locale parameter.

> Currently, the parameter is used once to translate the page, and then forgotten. The cookie is only set when selecting a different locale. It should also be set when loading the page.

**Comment by robotdan** [comment]
@glen-84 would love to hear if this solves your issue.

**Comment by glen-84** [comment]
@robotdan

Unfortunately it doesn't appear to work. I don't see any locale cookie being set.

Are you relying on the presence of the locale selection widget? If so, we don't use that. IMO, the cookie should be set on the server side.

We're using this as a workaround:

```js
// Set locale cookie.
const locale = new URLSearchParams(window.location.search).get("locale");

if (locale !== null) {
    document.cookie = `fusionauth.locale=${locale}; path=/`;
}

```

**Comment by robotdan** [comment]
Yes, this cookie is only written when using the widget. Your workaround seems reasonable. 

I forget why we did not set this on the server side, we can revisit this and see if we can figure out a better strategy to remove the requirement to use this widget.

**Comment by robotdan** [comment]
Moving to a new issue so we can leave this one closed since I delivered a change in 1.39.0.
- https://github.com/FusionAuth/fusionauth-issues/issues/1890
